mysql/resource_user.go

```diff
@@ -36,20 +36,50 @@ func resourceUser() *schema.Resource {
 				Sensitive: true,
 				StateFunc: hashSum,
 			},
+
 			"password": &schema.Schema{
 				Type:          schema.TypeString,
 				Optional:      true,
 				ConflictsWith: []string{"plaintext_password"},
 				Sensitive:     true,
 				Deprecated:    "Please use plaintext_password instead",
 			},
+
+			"auth": &schema.Schema{
```

Comment:
Can this be modified to `auth_plugin`, and make to be a flat `TypeString` value? I'm not too sure if we will ever be putting anything else in here - grants are already handled by `mysql_grant`, and we already have separate fields for plaintext passwords. So having the extra layer here is not entirely useful as it stands right now.

The field should also be made to conflict with all other authentication fields (including `password` as well, even though it's deprecated, it can still be used).